plugins/samples/check_pii/plugin.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_check_pii]
+package main
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/proxy-wasm/proxy-wasm-go-sdk/proxywasm"
+	"github.com/proxy-wasm/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+func main() {}
+func init() {
+	proxywasm.SetVMContext(&context{})
+}
+
+type context struct {
+	types.DefaultVMContext
+	types.DefaultPluginContext
+	creditCardRegex *regexp.Regexp
+}
+type httpContext struct {
+	types.DefaultHttpContext
+	creditCardRegex *regexp.Regexp
+	checkBody       bool
+}
+
+func (*context) NewPluginContext(contextID uint32) types.PluginContext {
+	return &context{creditCardRegex: regexp.MustCompile("\\d{4}-\\d{4}-\\d{4}-(\\d{4})")}
+}
+func (pluginContext *context) NewHttpContext(contextID uint32) types.HttpContext {
```

Comment:
Please add this caveat somewhere: https://github.com/GoogleCloudPlatform/service-extensions/blob/6170c6b99aac4fc4cd306dc58544eaf173b44ca4/plugins/samples/check_pii/plugin.cc#L37-L39

This code does not handle PII that spans chunk boundaries.